plugins/groups_manager.py

```diff
@@ -0,0 +1,307 @@
+                             help=_('make the group user visible (default)'))
+        visible.add_argument('--not-user-visible', dest='user_visible', action='store_false',
+                             default=None,
+                             help=_('make the group user invisible'))
+
+        # package list options
+        section = parser.add_mutually_exclusive_group()
+        section.add_argument('--mandatory', action='store_true',
+                             help=_('add packages to the mandatory section'))
+        section.add_argument('--optional', action='store_true',
+                             help=_('add packages to the optional section'))
+        section.add_argument('--remove', action='store_true', default=False,
+                             help=_('remove packages from group instead of adding them'))
+        parser.add_argument('--dependencies', action='store_true',
+                            help=_('include also direct dependencies for packages'))
+
+        parser.add_argument("packages", nargs='*', metavar='PACKAGE',
+                            help=_('package specification'))
+
+    def configure(self):
+        demands = self.cli.demands
+
+        if self.opts.packages:
+            demands.sack_activation = True
+            demands.available_repos = True
+            demands.load_system_repo = False
+
+        # handle --merge option (shortcut to --load and --save the same file)
+        if self.opts.merge:
+            self.opts.load.insert(0, self.opts.merge)
+            self.opts.save.append(self.opts.merge)
+
+        # check that group is specified when editing is attempted
+        if (self.opts.description
+                or self.opts.display_order
+                or self.opts.translated_name
+                or self.opts.translated_description
+                or self.opts.user_visible is not None
+                or self.opts.packages):
+            if not self.opts.id and not self.opts.name:
+                raise dnf.cli.CliError(
+                    _("Can't edit group without specifying it (use --id or --name)"))
+
+    def load_input_files(self):
+        """
+        Loads all input xml files.
+        Returns True if at least one file was successfuly loaded
+        """
+        for file_name in self.opts.load:
+            file_comps = libcomps.Comps()
+            try:
+                if file_name.endswith('.gz'):
+                    # libcomps does not support gzipped files - decompress to temporary
+                    # location
+                    with gzip.open(file_name) as gz_file:
+                        with tempfile.NamedTemporaryFile() as temp_file:
+                            shutil.copyfileobj(gz_file, temp_file)
+                            # The xml written to temp_file  might remain in buffers.
+                            # Force writing it to disk.
+                            temp_file.file.flush()
+                            os.fsync(temp_file.file.fileno())
```

Comment:
Shouldn't you close the file instead? I.e. do the reading on the line below out of the `with` block?

Also, shouldn't the `temp_file` be cleaned up? I would guess you can probably do it straight after loading it? (Regardless of whether the loading was successful)

Reply:
I used `with` block so that I do not need to care about handling error states, closing files and removing the temporary one - it is all provided by the context managers. That is also why I cannot do the reading out of the `with` block.

Reply:
Ok, I see. It's a bit awkward, the fact you need to do two calls and the second one being kind-of non-obvious doesn't make me too confident about this..

Reply:
I've changed flushing the file to closing it. Does this looks better?

Reply:
Actually, why not just use `fromxml_str()`? I guess that does mean loading the whole contents into memory... Is that an issue here?

What I don't like abou the new solutoin is using the context manager and then just circumventing it for the closing/unlinking... so if you wanna go this way, maybe just not use the context manager at all? Sorry about the back and forth...

Reply:
Exactly. I wanted to avoid loading the whole file into memory. I'll remove the inner context manager since is really doesn't do much.

Reply:
Thanks, I think this is better than the original :+1: